File: TwoNumberSum.cpp
#include <vector>
#include <algorithm>

using namespace std;

vector<int> twoNumberSum(vector<int> array, int targetSum) {
		std::sort(array.begin(), array.end());
		int leftPointer = 0;
		int rightPointer = array.size() - 1;
		while(leftPointer != rightPointer)
		{
			int currentSum = array[leftPointer] + array[rightPointer];
			if(currentSum == targetSum)
			{
				return {array[leftPointer], array[rightPointer]};
			}
			else if(currentSum < targetSum)
			{
				leftPointer++;
			}
			else if (currentSum > targetSum)
			{
				rightPointer--;
			}
		}
		return {};
}

File: algoexpert/Java/TwoNumberSum.java
import java.util.*;
class Program {
  public static int[] twoNumberSum(int[] array, int targetSum) {
    Arrays.sort(array);
		int leftHandPointer = 0;
		int rightHandPointer = array.length - 1;
		while(leftHandPointer < rightHandPointer)
		{
			int summ =  array[leftHandPointer] + array[rightHandPointer];
			if(summ == targetSum)
			{
				return new int[] {array[leftHandPointer], array[rightHandPointer]};
			}
			else if( summ > targetSum)
			{
				rightHandPointer--;
			}
			else
			{
				leftHandPointer++;
			}
		}
		return new int[0];
  }
}
